### PR TITLE
Be able to use unique language strings for the rules field

### PR DESCRIPTION
**Pull request by mbabker**
### Summary of Changes

Right now in Joomla core, the way to override a language string is to define the key in either the designated overrides file or load it in a file loaded after the version you want to override.  This is problematic in some cases because a key gets used in multiple contexts and if you were to load files in the wrong order this results in the wrong translation being displayed.  Additionally, this behavior makes it really difficult to create a process for caching a `JLanguage` instance with all of the language strings loaded into memory (you replace the processing time loading INIs into memory by just starting with the object having everything).

`JText` already exposes an API for using alternate language strings; it should be used in locations where we know localized contexts of a string may be created.  This PR does just that for the two keys in the rules form field that are commonly overridden by extensions. 

### Testing Instructions

In the com_content component config and article edit view, the permissions strings should use the localized keys for this component.

### Documentation Changes Required

N/A

### Language B/C Concerns

Though these keys are renamed in this PR, there is no B/C concern with translations.  `JText::alt()` checks if the alternate key (built by appending the second parameter to the first parameter) exists before using it, so if it doesn't exist, the base key (which will probably already be translated on a current 3.6 package) will be used.

### CODE REVIEW

**Comment by infograf768** [comment]
@mbabker 
we are using 22 instances of `JLIB_RULES_SETTING_NOTES` in core. (took off the lib ones)

Does it mean this PR should be enlarged to all these places?

**Comment by mbabker** [comment]
I didn't want to waste time doing them all if the idea was going to be rejected, I also wanted to have proof that this wasn't going to break anything.  If it's fine though I can work through them all.

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on 9ad428168d14b34007222b8cd3bedb0c727c389b<br /><br />Patch ok for me<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13523">issues.joomla.org/tracker/joomla-cms/13523</a>.</sub>

**Comment by jeckodevelopment** [comment]
@mbabker can you please look at the conflicts?

**Comment by infograf768** [comment]
folks, i think'it is a goog idea. let's keep it, if b/c, for 3.8

**Comment by franz-wohlkoenig** [comment]
@mbabker as Release Lead: Milestone Joomla 3.8.0?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13523">issues.joomla.org/tracker/joomla-cms/13523</a>.</sub>

**Comment by mbabker** [comment]
Rebased for 3.8.  I did all the strings except the one in `com_categories`, with this patch to have a unique string we would have to duplicate it in every extension supporting categories (the key would be `JLIB_RULES_SETTING_NOTES_ITEM_<extension>_CATEGORY` or we hardcode some rule into the form field to have yet another lookup for that context, doesn't seem worth the effort to do that.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 6e895de2033402e975ac244308ddfec0fbdb491f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13523">issues.joomla.org/tracker/joomla-cms/13523</a>.</sub>

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on 6e895de2033402e975ac244308ddfec0fbdb491f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13523">issues.joomla.org/tracker/joomla-cms/13523</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

File: administrator/language/en-GB/en-GB.com_content.ini [hunk 1]
```diff
@@ -202,5 +202,14 @@ COM_CONTENT_XML_DESCRIPTION="Article management component."
...
+
+; Alternate language strings for the rules form field
+JLIB_RULES_SETTING_NOTES_COM_CONTENT="Changes apply to this component only.<br /><em><strong>Inherited</strong></em> - a Global Configuration setting or higher level setting is applied.<br /><em><strong>Denied</strong></em> always wins - whatever is set at the Global or higher level and applies to all child elements.<br /><em><strong>Allowed</strong></em> will enable the action for this component unless it is overruled by a Global Configuration setting."
+JLIB_RULES_SETTING_NOTES_ITEM_COM_CONTENT_ARTICLE="Changes apply to this article only.<br /><em><strong>Inherited</strong></em> - a Global Configuration setting or higher level setting is applied.<br /><em><strong>Denied</strong></em> always wins - whatever is set at the Global or higher level and applies to all child elements.<br /><em><strong>Allowed</strong></em> will enable the action for this component unless it is overruled by a Global Configuration setting."
+
+COM_CONTENT_FIELDS_TYPE_MODAL_ARTICLE="Article"
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.com_content.ini]: hmm it look like this strings are dublicated?
https://github.com/joomla/joomla-cms/blob/6de2b00858e91bd64e7c410a6ed617d455f62d64/administrator/language/en-GB/en-GB.com_content.ini#L114-L117

Here are the strings in staging.

Comment (infograf768) [reply administrator/language/en-GB/en-GB.com_content.ini]: Is'nt it only
COM_CONTENT_FIELDS_TYPE_MODAL_ARTICLE="Article"

**Comment by infograf768** [comment]
Not sure this should go in 3.8.6. It can safely wait 3.9

**Comment by mbabker** [comment]
Rebased to 3.9.

**Comment by zero-24** [comment]
cc @wilsonge for 3.9. Drone is happy now too 😄

**Comment by mbabker** [comment]
This has been RTC for 10 months.  Is this getting merged or not?

**Comment by infograf768** [comment]
You are the one in charge. No opposition here for merging in 3.10

**Comment by brianteeman** [comment]
@mbabker is following best practice and not merging his own pr.

**Review by brianteeman — APPROVED** [review]
Approve language changes

**Comment by infograf768** [comment]
> @mbabker is following best practice and not merging his own pr.

@brianteeman 
Please refrain from posting when it is unnecessary. Also "approving" changes after the merge is quite useless, specially when you already have set your test as OK. 
@mbabker is big boy enough to ping maintainers in our glip.

**Comment by brianteeman** [comment]
I approved the change because after the last comment it triggered the notification from the code owners file so I had to approve it or it could not be merged. If you actually read instead of searching for faults you would see it was approved by me one hour before the merge.so get stop your pathetic little ranting and go outside and enjoy the summer sunshine

**Pull request merged**